[Bug] Failed to build BE with MySQL support

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?

```shell
ld.lld: error: duplicate symbol: SHA224
>>> defined at my_sha2.cc.o:(SHA224) in archive /Programs/doris/thirdparty/installed/lib/libmysqlclient.a
>>> defined at sha256.c
>>>            sha256.o:(.text+0x4F0) in archive /Programs/doris/thirdparty/installed/lib/libcrypto.a

ld.lld: error: duplicate symbol: SHA256
>>> defined at my_sha2.cc.o:(SHA256) in archive /Programs/doris/thirdparty/installed/lib/libmysqlclient.a
>>> defined at sha256.c
>>>            sha256.o:(.text+0x580) in archive /Programs/doris/thirdparty/installed/lib/libcrypto.a

ld.lld: error: duplicate symbol: SHA384
>>> defined at my_sha2.cc.o:(SHA384) in archive /Programs/doris/thirdparty/installed/lib/libmysqlclient.a
>>> defined at sha512.c
>>>            sha512.o:(.text+0x7D0) in archive /Programs/doris/thirdparty/installed/lib/libcrypto.a

ld.lld: error: duplicate symbol: SHA512
>>> defined at my_sha2.cc.o:(SHA512) in archive /Programs/doris/thirdparty/installed/lib/libmysqlclient.a
>>> defined at sha512.c
>>>            sha512.o:(.text+0x870) in archive /Programs/doris/thirdparty/installed/lib/libcrypto.a
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### What You Expected?

Build BE with MySQL support successfully.

### How to Reproduce?

```
WITH_MYSQL=ON ./build.sh --be --clean
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/cmake/thirdparty.cmake
+++ b/be/cmake/thirdparty.cmake
@@ -74,10 +74,6 @@ if(WITH_LZO)
     add_thirdparty(lzo LIBNAME "lib/liblzo2.a")
 endif()
 
-if (WITH_MYSQL)
-    add_thirdparty(mysql LIBNAME "lib/libmysqlclient.a")
-endif()
-
 add_thirdparty(libevent LIBNAME "lib/libevent.a")
 add_thirdparty(libevent_pthreads LIBNAME "lib/libevent_pthreads.a")
 add_thirdparty(libbz2 LIBNAME "lib/libbz2.a")
@@ -87,6 +83,10 @@ add_thirdparty(openssl LIBNAME "lib/libssl.a")
 add_thirdparty(leveldb)
 add_thirdparty(jemalloc LIBNAME "lib/libjemalloc_doris.a")
 add_thirdparty(jemalloc_arrow LIBNAME "lib/libjemalloc.a")
+
+if (WITH_MYSQL)
+    add_thirdparty(mysql LIBNAME "lib/libmysqlclient.a")
+endif()
 
 if (USE_UNWIND)
     add_thirdparty(libunwind LIBNAME "lib64/libunwind.a")

